ykrt/src/mt.rs

```diff
@@ -629,57 +649,73 @@ impl Drop for MT {
     }
 }
 
+/// [MTThread]'s major job is to record what state in the "interpreting/tracing/executing"
+/// state-machine this thread is in. This enum contains the states.
+enum MTThreadState {
+    /// This thread is executing in the normal interpreter: it is not executing a trace or
+    /// recording a trace.
+    Interpreting,
+    /// This thread is recording a trace.
+    Tracing {
+        /// The [HotLocation] the trace will end at. For a top-level trace, this will be the same
+        /// [HotLocation] the trace started at; for a side-trace, tracing started elsewhere.
+        hl: Arc<Mutex<HotLocation>>,
+        /// What tracer is being used to record this trace? Needed for trace mapping.
+        thread_tracer: Box<dyn TraceRecorder>,
+        /// Records the content of data recorded via `yk_promote`.
+        promotions: Vec<usize>,
+    },
+    /// This thread is executing a trace. Note that the `dyn CompiledTrace` serves two different purposes:
+    ///
+    /// 1. It is needed for side traces and the like.
+    /// 2. It allows another thread to tell whether the thread that started tracing a [Location] is
+    ///    still alive or not by inspecting its strong count (if the strong count is equal to 1
+    ///    then the thread died while tracing). Note that this relies on thread local storage
+    ///    dropping the [MTThread] instance and (by implication) dropping the [Arc] and
+    ///    decrementing its strong count. Unfortunately, there is no guarantee that thread local
+    ///    storage will be dropped when a thread dies (and there is also significant platform
+    ///    variation in regard to dropping thread locals), so this mechanism can't be fully relied
+    ///    upon: however, we can't monitor thread death in any other reasonable way, so this will
+    ///    have to do
```

Comment:
Missing `.`. Only fix this if you're pushing other commits anyway.

Reply:
Fixed in e0042c05ebc90f7c65df54bfb205e7963d4ff796.